client/src/app/components/header/header.component.html

```diff
@@ -0,0 +1,64 @@
+<div class="header-container">
+  <div class="logo-and-text">
+    <figure>
+      <img id="logo" src="../../../assets/Logo.png" alt="Money-Tree logo" />
+      <figcaption>Money Tree</figcaption>
+    </figure>
+  </div>
+
+  <div class="search-box-container">
+    <input id="searchbox" type="text" placeholder="Search for stocks ..." />
+  </div>
+
+  <div class="right-side-buttons">
+    <mat-icon routerLink="/" routerLinkActive="active" routerLinkActive="active">home</mat-icon>
+    <mat-icon routerLink="/leaderboard" routerLinkActive="active" routerLinkActive="active">leaderboard</mat-icon>
+    <a id="log-out-link" routerLink="/home" routerLinkActive="active">Log out</a>
+  </div>
+</div>
+
+<!-- when user is browsing on a mobile device -->
+<div class="header-container-mobile">
+  <button mat-icon-button [matMenuTriggerFor]="menu" class="menu-btn">
+    <mat-icon id="menu-btn-icon">menu</mat-icon>
+  </button>
+  <div class="logo-and-text-mobile">
+    <figure>
+      <img id="logo" src="../../../assets/Logo.png" alt="Money-Tree logo" />
+      <figcaption>Money Tree</figcaption>
+    </figure>
+  </div>
+
+  <mat-menu #menu="matMenu" class="mobile-menu-container">
+    <figure>
+      <img id= 'user-photo' src="../../../assets/user-photo.png" alt="user-photo" />
+      <figcaption class="figcaption-in-menu">
+        Hello Laurence! <br>
+        <!-- TODO: Change 'laurence' to the username of the user who logged in-->
+        <span id="score">Score:</span><span id="score-value">260</span>
+      </figcaption>
+    </figure>
+    <div class="menu-items">
+      <button mat-menu-item routerLink="/" routerLinkActive="active" routerLinkActive="active">
+        <mat-icon id="menu-item-icon" >home</mat-icon>
+        <span>Home</span>
+      </button>
+      <button mat-menu-item routerLink="/leaderboard" routerLinkActive="active" routerLinkActive="active">
+        <mat-icon id="menu-item-icon">
+          leaderboard</mat-icon>
+        <span>LeaderBoard</span>
+      </button>
+      <button mat-menu-item>
+        <mat-icon id="menu-item-icon">edit</mat-icon>
+        <span>Edit Profile</span>
+      </button>
+      <button mat-menu-item>
+        <mat-icon id="menu-item-icon">trending_up</mat-icon>
+        <span>Your Profile</span>
+      </button>
+      <button mat-menu-item routerLink="/home" routerLinkActive="active">
+        <a id="log-out-mobile">Log out</a>
+      </button>
```

Comment:
I know it's not what's in the sigma prototype, but I think the mobile logout button should be in the same list / style as the other buttons